### PR TITLE
langchain/libs/partners/google-genai/langchain_google_genai: safety_settings added to ChatGoogleGenerativeAI model in the langchain_google_genai library

### PR DESCRIPTION
**Pull request by 7vansh7**
- **Description** -> This PR **_adds safety_settings parameter_** to the **_ChatGoogleGenerativeAI_** class. The safety_settings parameter is originally there in the google.generativeai library's GenerativeModel class
- **Issue** -> https://github.com/langchain-ai/langchain/issues/15095
- **Dependencies** -> None added

### CODE REVIEW

**Comment by efriis** [comment]
Hey there! Can help more after the holidays, and looks like this needs some lint/format fixes.

File: libs/partners/google-genai/langchain_google_genai/chat_models.py [hunk 1]
```diff
@@ -509,6 +521,7 @@ def validate_environment(cls, values: Dict) -> Dict:
         values["client"] = genai.GenerativeModel(model_name=model)
```
Comment (jimwhite) [inline libs/partners/google-genai/langchain_google_genai/chat_models.py:521]: I think the place to use the model instance (i.e. self) safety_settings is here when the client is created:
safety_settings = values.get("safety_settings")  # I assume missing key is None value
genai.GenerativeModel(model_name=model, safety_settings=safety_settings)

**Comment by jimwhite** [comment]
Great to see this in progress because I've just been dealing with this too, and I hate to slow things down but I don't think the model instance self._safety_settings should be used on the chat calls (they can get per call kwargs in the future).  As I commented in the code I think this __init__ param should be applied when the client is created by the genai.GenerativeModel call.

**Review by rayanfer32 — APPROVED** [review]
Changes work 👍 you can test it by passing 
```py
safety_settings = [
  {
    "category": "HARM_CATEGORY_HARASSMENT",
    "threshold": "BLOCK_ONLY_HIGH"
  },
  {
    "category": "HARM_CATEGORY_HATE_SPEECH",
    "threshold": "BLOCK_ONLY_HIGH"
  },
  {
    "category": "HARM_CATEGORY_SEXUALLY_EXPLICIT",
    "threshold": "BLOCK_ONLY_HIGH"
  },
  {
    "category": "HARM_CATEGORY_DANGEROUS_CONTENT",
    "threshold": "BLOCK_ONLY_HIGH"
  },
]
gemini_llm = ChatGoogleGenerativeAI(
    model="gemini-pro", verbose=True, temperature=0.1, google_api_key=api_gemini, safety_settings=safety_settings
)
```

**Comment by izapolsk** [comment]
there is another PR doing the same but it has tests and etc 
https://github.com/langchain-ai/langchain/pull/15344

**Comment by efriis** [comment]
#16836

**Pull request closed without merging**